### PR TITLE
Verification pipeline boilerplate

### PR DESCRIPTION
**Pull request by delucchi-cmu**
## Change Description

- [x] My PR includes a link to the issue that I am addressing

Adds the boilerplate for running the new verification pipeline within the existing hipscat-import pipeline infrastructure.

This allows us to inherit:
- dask client creation
- success/failure email
- creation and management of output path and temporary path
- basic provenance tracking

## Code Quality
- [x] I have read the [Contribution Guide](https://lincc-ppt.readthedocs.io/en/latest/source/contributing.html)
- [x] My code follows the code style of this project
- [x] My code builds (or compiles) cleanly without any errors or warnings
- [x] My code contains relevant comments and necessary documentation

### New Feature Checklist
- [x] I have added or updated the docstrings associated with my feature using the [NumPy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html)

### CODE REVIEW

**Review by troyraen — COMMENTED** [review]
I'm out of the office unexpectedly, but plan to be back this Thursday, Sept 7. I can take a closer look when I return. In the meantime, can you give me some direction about what kind of review you'd like? I gather that you'd like me to write the code that run_verification.run will execute? If so, I can look at this PR from the perspective of understanding how to do that. Are there other things you'd like me to look at?

**Comment by delucchi-cmu** [comment]
Our general guidelines and philosophy on code reviews are in this wiki entry: https://github.com/lincc-frameworks/docs/wiki/Design-and-Code-Review-Policy

I'm hoping that by getting this boilerplate PR out of the way, adding the meaningful implementation of the verification logic will be more straightforward. Your assumption of the perspective I'm looking for is correct, though I'm planning to also contribute some to the logic in `run_verification` - you won't be out on your own =]

File: src/hipscat_import/verification/arguments.py
Comment (troyraen) [inline src/hipscat_import/verification/arguments.py:20]: Is there a valid case where this could contain anything other than the RA and dec column names?

**Review by troyraen — APPROVED** [review]
Some questions and comments below, but overall LGTM and I'll approve and let you decide when you're ready to merge.

File: src/hipscat_import/verification/arguments.py
Comment (delucchi-cmu) [reply src/hipscat_import/verification/arguments.py:15]: "input" here means the hipscat catalog. "output" is any text or images resulting from inspecting the catalog and data. We're inheriting a lot of nice functionality in the `RuntimeArguments`, though I agree the naming isn't ideal. Maybe `output_artifact_path`? Either way, that would be a significant code change, so I'd want to do it outside of this PR.

File: src/hipscat_import/verification/arguments.py
Comment (delucchi-cmu) [reply src/hipscat_import/verification/arguments.py]: The idea is that a user can just provide the string path to the catalog, and not need to load the in-memory representation. I've made the options a little clearer with some docstrings, and accepting either as valid parameters.

File: src/hipscat_import/verification/arguments.py
Comment (delucchi-cmu) [reply src/hipscat_import/verification/arguments.py:23]: I can imagine wanting to see the distribution of any field that has some expected boundaries: flux/flux_err, mag/mag_err, time/mjd, band. I don't think we'd want to produce histograms for all values, however, since this would potentially also include numeric ids.

File: src/hipscat_import/verification/arguments.py
Comment (delucchi-cmu) [reply src/hipscat_import/verification/arguments.py]: Updated.

File: tests/hipscat_import/verification/test_run_verification.py
Comment (delucchi-cmu) [reply tests/hipscat_import/verification/test_run_verification.py:22]: See above. We could consider `output_artifact_path`, since the pipelines in this package vary a lot in the types of data they output.

File: tests/hipscat_import/verification/test_verification_arguments.py
Comment (delucchi-cmu) [reply tests/hipscat_import/verification/test_verification_arguments.py:1]: We make heavy use of pytest fixtures, which set up various resources used for unit testing, or provide consistent paths. These are defined in `conftest.py`. They're powerful and flexible (and fun in their own way), but I can appreciate the learning curve involved. 

I'll expand the development guide to include the below link, as well as the one I sent you on LINCC Frameworks design and code reviews.

https://docs.pytest.org/en/7.1.x/explanation/fixtures.html#about-fixtures

I'm curious if you have other thoughts on surfacing repo-specific conventions.

File: tests/hipscat_import/verification/test_verification_arguments.py
Comment (delucchi-cmu) [reply tests/hipscat_import/verification/test_verification_arguments.py:33]: The `small_sky_object_catalog` referenced here is a pytest fixture, and injected at runtime. It would be cute if we could do something like:

```
def test_something(specific_use=generic_fixture_value):
  ...
```

Though I guess an alternative convention might be:

```
def test_something(generic_fixture_value):
  specific_use = generic_fixture_value
  ...
```

File: tests/hipscat_import/verification/test_verification_arguments.py
Comment (troyraen) [reply tests/hipscat_import/verification/test_verification_arguments.py:1]: Chalk this up to my lack of familiarity with pytest. By the way, I didn't realize before I did the review that the repo was set to require all conversations to be resolved before merging. Whenever you want to just mark them all resolved and merge is totally fine with me. I don't mean for anything I'm saying to be a blocker -- more just part of my learning process.

> I'm curious if you have other thoughts on surfacing repo-specific conventions.

I am a big fan of self-documenting code. External guides are good for some things but when I'm trying to read and understand code I find it difficult to use such a guide effectively because it's really hard to know where to look when a specific question pops up. On the other hand, the API docs are my go-to; I read those all the time. I much prefer for all the info I need to understand a piece of code to be in the code itself (clear class/function/variable names, type hints, logical structure broken into smallish discrete chunks and a flow between them that's easy to follow, etc.) or right next to it (docstrings, code comments where it's hard to make things clear enough with names and logical structure alone).

**Pull request merged**
